engine/runtime/src/main/scala/org/enso/compiler/core/IR.scala

```diff
@@ -726,13 +726,32 @@ object IR {
             * @param packageName the name of the package containing the imported
             *                    class
             * @param className the class name
+            * @param rename the name this object should be visible under in the
+            *               importing scope
             */
-          case class Java(packageName: String, className: String)
-              extends Entity {
+          case class Java(
+            packageName: String,
+            className: String,
+            rename: Option[String]
```

Comment:
The rename should be a property of the `Polyglot` import, not specific to polyglot java imports.